### ISSUE TITLE
cmd/go: list -e -json [absolute directory outside gopath] fails

### ISSUE THREAD

**Issue by matloob** [golang/go#34273]
### What version of Go are you using (`go version`)?

<pre>
$ go version
go version go1.13 darwin/amd64
</pre>

### Does this issue reproduce with the latest release?

yes

### What operating system and processor architecture are you using (`go env`)?

<details><summary><code>go env</code> Output</summary><br><pre>
$ go env

GO111MODULE="on"
GOARCH="amd64"
GOBIN=""
GOCACHE="/Users/matloob/Library/Caches/go-build"
GOENV="/Users/matloob/Library/Application Support/go/env"
GOEXE=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="darwin"
GONOPROXY=""
GONOSUMDB=""
GOOS="darwin"
GOPATH="/Users/matloob/go"
GOPRIVATE=""
GOPROXY="https://proxy.golang.org,direct"
GOROOT="/usr/local/go"
GOSUMDB="sum.golang.org"
GOTMPDIR=""
GOTOOLDIR="/usr/local/go/pkg/tool/darwin_amd64"
GCCGO="gccgo"
AR="ar"
CC="clang"
CXX="clang++"
CGO_ENABLED="1"
GOMOD="/Users/matloob/tools/go.mod"
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fno-caret-diagnostics -Qunused-arguments -fmessage-length=0 -fdebug-prefix-map=/var/folders/f4/9p58ddnj40x58zchdb36p7lr004_sl/T/go-build066579511=/tmp/go-build -gno-record-gcc-switches -fno-common"
</pre></details>

### What did you do?

1. set `GO111MODULE=on`
2. change current directory to a module
3. run `go list -e -json /Users/matloob/Desktop` (or list any existing directory outside of any module)

### What did you expect to see?

exit status of 0, empty stderr, and a "Fake" package with errors set to a message saying the directory is outside any modules.

### What did you see instead?

exit status of 1 and stderr of "go: directory ../Desktop outside available modules\n"

**Comment by bcmills** [comment]
Possibly related: #29280, #26909.

**Comment by dominikh** [comment]
Another variation of this error:

```
$ go list ../...
go: pattern ../... refers to dir /tmp/foo, outside module root /tmp/foo/b
```

**Comment by dominikh** [comment]
I believe this is fixed in Go 1.15:

```
$ go version
go version go1.15beta1 linux/amd64
$ go list -json -e ../...
{
	"ImportPath": "../...",
	"Match": [
		"../..."
	],
	"Incomplete": true,
	"Error": {
		"ImportStack": [],
		"Pos": "",
		"Err": "pattern ../...: directory prefix .. outside available modules"
	}
}
$ echo $?
0
$ go list -e -json /home/dominikh/prj/src/sandbox/bar
{
	"ImportPath": "/home/dominikh/prj/src/sandbox/bar",
	"Match": [
		"/home/dominikh/prj/src/sandbox/bar"
	],
	"Incomplete": true,
	"Error": {
		"ImportStack": [],
		"Pos": "",
		"Err": "directory ../bar outside available modules"
	}
}
$ echo $?
0
```

`git bisect` points to one of these commits: 156c60709e7775a6f1baa9685155b5974fefdea9, d11e1f92fc578c5d2e604acfe9ea60d7afb84a0c, 5a61de3fe160cc8b327ee893cd74c4d0ce9dc13d

@matloob can you confirm this is now working as intended?

**Comment by matloob** [comment]
Yeah, it's now working as intended. Closing.
